app/src/main/res/layout/bgsource_item.xml

```diff
@@ -39,52 +37,52 @@
                 android:id="@+id/time"
                 android:layout_width="wrap_content"
                 android:layout_height="wrap_content"
-                android:text="16:55"
-                android:textStyle="bold" />
+                android:textStyle="bold"
+                tools:text="16:55" />
 
             <TextView
                 android:id="@+id/value"
                 android:layout_width="wrap_content"
                 android:layout_height="wrap_content"
                 android:paddingStart="10dp"
-                android:text="Name"
-                android:textStyle="bold" />
+                android:textStyle="bold"
+                tools:text="Name" />
 
             <ImageView
                 android:id="@+id/direction"
                 android:layout_width="wrap_content"
                 android:layout_height="match_parent"
                 android:layout_gravity="center_vertical"
-                card_view:srcCompat="@drawable/ic_flat" />
+                app:srcCompat="@drawable/ic_flat" />
 
             <TextView
                 android:id="@+id/ns"
                 android:layout_width="wrap_content"
                 android:layout_height="wrap_content"
                 android:paddingStart="10dp"
-                android:text="NS"
                 android:textAlignment="viewEnd"
-                android:textColor="?attr/setTempButton" />
+                android:textColor="?attr/setTempButton"
+                tools:text="NS" />
 
             <TextView
                 android:id="@+id/invalid"
                 android:layout_width="wrap_content"
                 android:layout_height="wrap_content"
                 android:paddingStart="10dp"
-                android:text="@string/invalid"
-                android:textColor="?attr/alarmColor" />
+                android:textColor="?attr/alarmColor"
```

Comment:
see this https://github.com/nightscout/AndroidAPS/commit/3d729820638f807f311e51e480f2c04a45ae23a0
tools:text is displayed only in IDE but not in app

Reply:
Indeed, your right. In this cases it should not just in the IDE. (same for NS label). Will fix it

Reply:
I see now how I broke it last time.